### PR TITLE
Airbyte CDK: Change Error message if stream is not found (Deprecated)

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*Resolve issue #27393*

## How
1. Change `error message`
2. Add property `raise_exception_on_missing_stream` to `AbstractSource`

## Recommended reading order
1. `airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py`
3. `airbyte-cdk/python/unit_tests/sources/test_abstract_source.py`

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-cdk/python/unit_tests/sources/test_abstract_source.py [hunk 1]
```diff
@@ -239,6 +245,21 @@ def test_read_nonexistent_stream_raises_exception(mocker):
...
+    mocker.patch.object(MockStream, "get_json_schema", return_value={})
+
+    src = MockSource(streams=[s1], exception_on_missing_stream=False)
+
+    catalog = ConfiguredAirbyteCatalog(streams=[_configured_stream(s2, SyncMode.full_refresh)])
+    messages = list(src.read(logger, {}, catalog))
+
+    assert messages == []
+
+
```
Comment (keu) [inline airbyte-cdk/python/unit_tests/sources/test_abstract_source.py:262]: let's add a test for `exception_on_missing_stream=True` as well

**Review by keu — APPROVED** [review]

**Review by girarda — APPROVED** [review]
this is great!

@sherifnada can you weight in on why some source might not want to raise an exception on a missing stream? It's not clear to me why skipping missing streams would be the expected behavior

**Comment by sherifnada** [comment]
@girarda some sources like FB marketing allow configuring streams in the spec (eg: custom reports). But if the user changes the config to remove that custom report and doesn't refresh the schema then the sync starts failing because of a missing stream. So those connectors would benefit form not raising exceptions on missing streams.

**Comment by yevhenii-ldv** [comment]
airbyte-ci connectors --name=source-stripe test

**Comment by girarda** [comment]
Thanks for the context @sherifnada!

@yevhenii-ldv let's update the facebook and google-ads connectors so they do not raise an exception on missing streams. this can be done in a separate PR after the CDK version has been bumped. 

Follow up issue: https://github.com/airbytehq/airbyte/issues/30701

**Pull request closed without merging**